add switch, radio input and file upload support for settings form

dvitcompte opened the pull request:
add switch, radio input and file upload support for settings form.

I'm not sure if file upload process contains all needed verifications (file size fo example).
Tested on ps 1.6

settings.json updated in email-templates-sdk for example

Scritik commented:
Some parts of your code are indented with tabs and not spaces. You forgot the method's visibility. Can you make sure that all your code is PSR-2? Thanks.

@rGaillard Can you take a look? Seems ok to me but I want to be sure since we have some uploads possibility. Thx.

dvitcompte commented:
@Scritik 

Hello!
My bad I forgot to launch the PSR-2 plugin. Now it's done.
I did not set the $max_file_size argument for the method ImageManager::validateUpload($file). I don't know if you want to set a limit and the value of this limit (500kB, 1Mb?). 
The file upload could be usefull to change little image like socials network, or add a background image in the header (I'm developping a template using a background header image) 

I also fixed the indent in the settings.json file [here](https://github.com/PrestaShop/email-templates-sdk/pull/5/files) (826fe58)

Thanks for the feedback!

Regards,

dvitcompte commented:
Hello, still no review for this pull request?
Thx

dvitcompte commented:
Hello, still no review for this pull request?
Thx

Scritik commented:
Hi @dvitcompte,

Sorry for the delay, I think it's time to get back to this module :)

I have a few questions/remarks:

- The support for a switch is interesting but in PS1.5, it's not supported if I remember. You should add a condition to transform the the switch into a radio button on 1.5. 
- `is_bool` is a bool, so the default value should be a bool :)
- Same for values, you should assign an empty array (just to keep the same type).

@rGaillard Can you take a look at `uploadFiles()`? thx.

Thanks.

dvitcompte commented:
Hello @Scritik ,
happy to read you again :)
I added a condition for PS15 to display a radio button instead of a switch. I also added a "class" property with the value 't' which is used for PS15 to display properly the radio buttons.

I'm preparing commits to add support for select field, and tabs (only for PS16 & PS17)

Regards

Scritik commented:
Thank you! I will take a look at it asap.

While select can be interesting, please do not make tabs if they are not compatible 1.5. I don't think it is really useful by the way, yes the form can be long but the choices are generally quite simple.

Thanks!

dvitcompte commented:
Hello,

I was wondering the same about tabs, but as PS16 and 17 are the most used and more and more shops are updated to PS17 or PS16. And for example my Minimalist Pro template has 29 options : 12 only for all socials networks, 8 for colors. I thought that classifying them by theme would be a good idea. 
I will use <hr> instead in my settings form :)

settings.json has been updated with an example of select field in email-templates-sdk project.

Regards

Scritik commented:
Yes they are the most used but this module is compatible with the 1.5 for now.
Keep this idea when we will work on a 2.0, only compatible with 1.6+.

And by the way, 12 social networks? Don't you think it's a little bit too much for merchants? ;)

dvitcompte commented:
Hello,

there are "only" 8 socials networks ;), a multilang field to add a custom title, 2 fields to add a custom social network, the last is to select the position of this block in the email. This is the most advanced module and others are more simple.

Scritik commented:
8 seems more reasonable :)

Scritik commented:
I started to test your PR and I found an issue. If I have a file input in the form and I edit the template, the file is not saved, and I need to upload a file every time!

A solution is to save the value like the other values, and if no file is uploaded, you get the saved value and use it instead of the empty value.
